### PR TITLE
Update specification for ConvolutionBackpropData.

### PR DESCRIPTION
**Pull request by sdurawa**
### Details:
 - update all sections of ConvolutionBackpropData specification

### Tickets:
 - 46091

### CODE REVIEW

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 1]
```diff
@@ -76,7 +76,8 @@ else:
 
   * **Description**: *auto_pad* has the same definition as *auto_pad* for a regular Convolution but applied in the backward way, for the output tensor. 
     * *explicit*: use explicit padding values from `pads_begin` and `pads_end`.
-    * *same_upper (same_lower)* the input is padded to match the output size. In case of odd padding value an extra padding is added at the end (at the beginning).
+    * *same_upper* the input is padded to match the output size. In case of odd padding value an extra padding is added at the end.
+    * *same_lower* the input is padded to match the output size. In case of odd padding value an extra padding is added at the beginning.
     * *valid* - do not use padding.
   * **Type**: string
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:82]: Please add backticks ("`") also for types in all attributes. ie: `**Type**: `string``

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 2]
```diff
@@ -93,21 +94,30 @@ else:
 
 **Inputs**:
 
-*   **1**: `data` -- input tensor of rank 3 or greater. Layout is `[N, C_INPUT, X1, ..., XD]`. *Required*.
+*   **1**: Input tensor of type *T1* and rank 3, 4 or 5. Layout is NCZYX (number of batches, number of channels, spatial axes Z, Y, X). *Required*.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:97]: I would leave the shape layout as it was, so we know explicitly that C dim corresponds to I dim of filter.
`[N, C_INPUT, X1, ..., XD]` or `[N, C_INPUT, Z, Y, X]` if you prefer

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 3]
```diff
@@ -93,21 +94,30 @@ else:
...
 
-*   **2**: `filter` -- convolution kernel tensor. Weights have shape `[C_INPUT, C_OUTPUT, K_D, ..., K_1]`. `C_INPUT` is the number of channels in input `data` tensor shape, and `C_OUTPUT` is the number of channels in the `output` tensor. Spatial size of the kernel `[K_D, ..., K_1]` is derived from the shape of this input and aren't specified by any attribute. *Required*.
+*   **2**: Convolution kernel tensor of type *T1* and rank 3, 4 or 5. Layout is IOZYX (number of input channels, number of output channels, spatial axes Z, Y, X). Spatial size of the kernel is derived from the shape of this input and aren't specified by any attribute. *Required*.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:99]: Removing shape layout may mislead the user. It looks like with respect to Convolution, the dimensions are reversed.
filters in Convolution have shape `[C_OUT, C_IN, Z, Y, X]`, while in Backprop the shape is `[C_INPUT, C_OUTPUT, K_D, ..., K_1]`. Also based on current reference implementation for ConvBackprop ([here](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp#L252)) I see that spatial dimensions of data input are reversed. @lazarevevgeny maybe you could give a comment here?

Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:99]: @dmitry-gorokhov could you explain how it works from CPU plugin point of view?

Comment (dmitry-gorokhov) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:99]: @sdurawa As you correctly mentioned CPU plug-in expects weights in [C_INPUT, C_OUTPUT, K_D, ..., K_1] layout.

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 4]
```diff
@@ -93,21 +94,30 @@ else:
...
+      * 3D convolution (input tensors rank 5) means that there are three spatial axes Z, Y, X
 
 **Outputs**:
 
-*   **1**: `output` -- output tensor of the same rank as input `data` tensor and shape `[N, C_OUTPUT, Y1, ..., YD]`.
+*   **1**: Output tensor of type *T1* and rank 3, 4 or 5. Layout is NOZYX (number of batches, number of kernel output channels, spatial axes Z, Y, X).
+
+**Types**:
+
+* *T1*: any floating point type.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:113]: Please change this line to support any numeric type

**Review by ggalieroc — COMMENTED** [review]
Please update the comments to this PR

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 1 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Fixed.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 2 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Yes, that is better solution to use existing layout.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 3 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Filter layout changed to the same as in cpu plugin.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 4 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: fixed.

**Comment by sdurawa** [comment]
@lazarevevgeny, may you review this op?

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 5]
```diff
@@ -87,27 +88,36 @@ else:
...
+
+* *T1*: any numeric type.
+* *T2*: any integer type.
 
 **Example**
 
 ```xml
 <layer id="5" name="upsampling_node" type="ConvolutionBackpropData">
-    <data dilations="1,1" pads_begin="1,1" pads_end="1,1" strides="2,2"/>
+    <data dilations="1,1" pads_begin="1,1" pads_end="1,1" strides="2,2", output_padding="0,0/>
```
Comment (lazarevevgeny) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:120]: incorrect xml. There should be no comma in between attributes and missing double quote in the end.

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 6]
```diff
@@ -4,15 +4,15 @@
 
 **Category**: Convolution
 
-**Short description**: Computes the gradients of a Convolution operation with respect to the input. Also known as a Deconvolution or a Transposed Convolution.
+**Short description**: Computes 1D, 2D or 3D *ConvolutionBackpropData* operation with respect to the input and kernel tensors. Also known as a Transposed Convolution.
```
Comment (lazarevevgeny) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:7]: Why did you remove mentioning of "Deconvolution"?

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 7]
```diff
@@ -4,15 +4,15 @@
...
+ConvolutionBackpropData accepts the same set of attributes as a regular Convolution operation, but they are interpreted in a "backward way", so they are applied to the output of ConvolutionBackpropData, but not to the input. Refer to a regular [Convolution](Convolution_1.md) operation for detailed description of each attribute.
 
-Output shape when specified as an input `output_shape`, specifies only spatial dimensions. No batch or channel dimension should be passed along with H, W or other spatial dimensions. If `output_shape` is omitted, then `pads_begin`, `pads_end` or `auto_pad` are used to determine output spatial shape `[Y_1, Y_2, ..., Y_D]` by input spatial shape `[X_1, X_2, ..., X_D]` in the following way:
+Output shape when specified as an input `output_shape`, specifies only spatial dimensions. No batch or channel dimension should be passed along with spatial dimensions. If `output_shape` is omitted, then `pads_begin`, `pads_end` or `auto_pad` are used to determine output spatial shape `[Y_z, Y_y, Y_x]` by input spatial shape `[X_z, X_y, X_x]` in the following way:
```
Comment (lazarevevgeny) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:15]: Change "Output shape when specified as an input \`output_shape\`, specifies only spatial dimensions." to "When output shape is specified as an input tensor \`output_shape\` then it specifies only spatial dimensions."

Comment (lazarevevgeny) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:15]: Names `Y_z, Y_y, Y_x, X_z, X_y, X_x` are really confusing... Maybe rename "X" to "I" and "Y" to "O" to reflect the fact that they correspond to Input and Output tensors?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 6 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:7]: Deconvolution is a mathematical operation which reverts process of convolution (on the output produces original image). Transposed convolution only reconstructs the spatial dimensions of the image, but output data is not the same as on the input. I removed Deconvolution term as it not exactly fits to this op.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 5 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:120]: Corrected.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 7 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Corrected.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 6 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:7]: @lazarevevgeny, please close this discussion if you are ok with my response.

**Review by lazarevevgeny — APPROVED** [review]

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 8]
```diff
@@ -87,27 +88,36 @@ else:
...
 
-*   **1**: `output` -- output tensor of the same rank as input `data` tensor and shape `[N, C_OUTPUT, Y1, ..., YD]`.
+*   **1**: Output tensor of type *T1* and rank 3, 4 or 5. Layout is `[N, C_OUTPUT, Z, Y, X]` (number of batches, number of kernel output channels, spatial axes Z, Y, X).
+
+**Types**:
+
+* *T1*: any numeric type.
+* *T2*: any integer type.
 
 **Example**
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:116]: I thinks we should add more examples including optional 3rd input output_shape

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 9]
```diff
@@ -87,27 +88,36 @@ else:
...
+*   **2**: Convolution kernel tensor of type *T1* and rank 3, 4 or 5. Layout is `[C_INPUT, C_OUTPUT, Z, Y, X]` (number of input channels, number of output channels, spatial axes Z, Y, X). Spatial size of the kernel is derived from the shape of this input and aren't specified by any attribute. *Required*.
 
-*   **3**: `output_shape` is 1D integer tensor that specifies spatial shape of the output. *Optional*. If specified, *padding amount* is deduced from relation of input and output spatial shapes according to formulas in the description. If not specified, *output shape* is calculated based on the `pads_begin` and `pads_end` or completely according to `auto_pad`.
+*   **3**: `output_shape` is 1D tensor of type *T2* that specifies spatial shape of the output. *Optional*. If specified, *padding amount* is deduced from relation of input and output spatial shapes according to formulas in the description. If not specified, *output shape* is calculated based on the `pads_begin` and `pads_end` or completely according to `auto_pad`.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:101]: Third input is optional. We should add \*\*Optional.\*\* at the end of the input description

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 10]
```diff
@@ -4,15 +4,15 @@
...
 
 **Detailed description**:
 
 ConvolutionBackpropData takes the input tensor, weights tensor and output shape and computes the output tensor of a given shape. The shape of the output can be specified as an input 1D integer tensor explicitly or determined by other attributes implicitly. If output shape is specified as an explicit input, shape of the output exactly matches the specified size and required amount of padding is computed.
 
-ConvolutionBackpropData accepts the same set of attributes as a regular Convolution operation, but they are interpreted in a "backward way", so they are applied to the output of ConvolutionBackpropData, but not to the input. Refer to a regular Convolution operation for detailed description of each attribute.
+ConvolutionBackpropData accepts the same set of attributes as a regular Convolution operation, but they are interpreted in a "backward way", so they are applied to the output of ConvolutionBackpropData, but not to the input. Refer to a regular [Convolution](Convolution_1.md) operation for detailed description of each attribute.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:13]: In fact the set of attributes is the same as Convolition + one more (output_padding). Shouldn't we mention this here too?

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 11]
```diff
@@ -4,15 +4,15 @@
 
 **Category**: Convolution
 
-**Short description**: Computes the gradients of a Convolution operation with respect to the input. Also known as a Deconvolution or a Transposed Convolution.
+**Short description**: Computes 1D, 2D or 3D *ConvolutionBackpropData* operation with respect to the input and kernel tensors. Also known as a Transposed Convolution.
 
 **Detailed description**:
 
 ConvolutionBackpropData takes the input tensor, weights tensor and output shape and computes the output tensor of a given shape. The shape of the output can be specified as an input 1D integer tensor explicitly or determined by other attributes implicitly. If output shape is specified as an explicit input, shape of the output exactly matches the specified size and required amount of padding is computed.
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:11]: How about adding some reference article about this operation as other related convolution operation do? E.g. https://arxiv.org/abs/1603.07285

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 8 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:116]: New examples are added.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 9 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: *optional* is already included after first sentence. I moved it to the end of input description.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 10 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Added specific information that output_padding is additional attribute.

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 11 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:11]: As far as I remember we shall limit linking to external resources. @lazarevevgeny shall we add link for detailed convolution backprop description?

Comment (ggalieroc) [reply docs/ops/convolution/ConvolutionBackpropData_1.md:11]: I would say only to unofficial external resources, but not to official research publications/papers. Please have a look to other convolution operations spec:

- [Convolution](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/Convolution_1.md)
- [BinaryConvolution](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/BinaryConvolution_1.md)
- [GroupConvolution](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/GroupConvolution_1.md)
- [DeformableConvolution](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/DeformableConvolution_1.md)

Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Ok, I've added link to arxiv article.

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 12]
```diff
@@ -87,27 +88,37 @@ else:
...
 
-*   **1**: `output` -- output tensor of the same rank as input `data` tensor and shape `[N, C_OUTPUT, Y1, ..., YD]`.
+*   **1**: Output tensor of type *T1* and rank 3, 4 or 5. Layout is `[N, C_OUTPUT, Z, Y, X]` (number of batches, number of kernel output channels, spatial axes Z, Y, X).
+
+**Types**:
+
+* *T1*: any numeric type.
+* *T2*: any integer type.
 
 **Example**
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:116]: nitpick: \*\*Examples\*\*

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 13]
```diff
@@ -87,27 +88,37 @@ else:
...
 
+2D ConvolutionBackpropData
```
Comment (ggalieroc) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:118]: Maybe enumerate examples same way as in [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/generation/Range_4.md)

**Review by ggalieroc — APPROVED** [review]
I am approving since it LGTM, just nitpick comments left

File: docs/ops/convolution/ConvolutionBackpropData_1.md [hunk 14]
```diff
@@ -132,3 +144,67 @@ else:
...
+        </port>
+        <port id="1">
+            <dim>20</dim>
+            <dim>10</dim>
+            <dim>3</dim>
+            <dim>3</dim>
+        </port>
+        <port id="2">
+            <dim>450</dim>
+            <dim>450</dim>
```
Comment (lazarevevgeny) [inline docs/ops/convolution/ConvolutionBackpropData_1.md:198]: Wrong example. The shape of the third input should be `[2]` and the **elements** in this tensor should be `[450, 450]`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/convolution/ConvolutionBackpropData_1.md (hunk 14 above)
Comment (sdurawa) [reply docs/ops/convolution/ConvolutionBackpropData_1.md]: Corrected.
